core/logger.py

```diff
@@ -0,0 +1,107 @@
+import datetime
+import json
+import logging
+import queue
+from logging import Formatter
+from logging.handlers import QueueHandler, QueueListener
+
+from core.schemas.audit import AuditLog
+
+# Inspired by 
+# * https://www.sheshbabu.com/posts/fastapi-structured-json-logging/
+# * https://rob-blackbourn.medium.com/how-to-use-python-logging-queuehandler-with-dictconfig-1e8b1284e27a
+
+
+class ArangoHandler(logging.Handler):
+
+    actions = {
+        "GET": "read",
+        "POST": "create",
+        "PATCH": "update",
+        "DELETE": "delete",
+    }
+
+    def __init__(self, level=logging.NOTSET):
+        super().__init__(level)
+
+    def emit(self, record):
+        if "type" not in record.__dict__:
+            return
+        if record.__dict__["type"] != "audit.log":
+            return
+        target = record.__dict__["path"]
+        if "/auth/" in target or target.endswith("/search"):
+            return
+        action = self.actions.get(record.__dict__["method"], "unknown")
+        if record.__dict__["status_code"] == 200:
+            status = "succeeded"
+        else:
+            status = "failed"
```

Comment:
Maybe we could also store the status code? So we can tell the difference between 401, 403, 50x

Reply:
Fixed in 34d2b00303af967901d72f529b80a89b8aff022c